wire-ios-sync-engine/Source/Synchronization/SyncAgent.swift

```diff
@@ -205,8 +197,15 @@ final class SyncAgent: NSObject, SyncAgentProtocol {
                 try await incrementalSyncTaskManager.performIfNeeded { [weak self] in
                     guard let self else { return }
                     delegate?.syncAgentDidStartIncrementalSync(self)
-                    incrementalSyncToken = try await incrementalSyncProvider.provideIncrementalSync().perform()
-                    delegate?.syncAgentDidFinishIncrementalSync(self, isRecovering: false)
+
+                    if liveSync {
+                        incrementalSyncToken = try await incrementalSyncProvider.provideLiveSync(delegate: self)
+                            .perform(acknowledgeFullSync: shouldAcknowledgeFullSync)
```

Comment:
in case of liveSync the incrementalSync will call the SyncAgent via LiveSyncDelegate when we're up to date and therefore hide the sync bar

**thought:** I guess at some point we could rely on the publisher of sync state instead of the delegates to interact with the sync bar